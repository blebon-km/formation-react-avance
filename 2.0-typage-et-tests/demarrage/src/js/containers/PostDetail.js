import React from 'react';
import { connect } from 'react-redux';
import config from 'config';
import { fetchPost } from '../actions';

class PostDetail extends React.Component {
    render() {
        const { data, isLoading } = this.props.post;
        return <section className={`post-detail ${isLoading ? "is-loading" : ""}`}>
            {
                data && (
					<div>
						<div className="picture-container">
							<img src={config.picturesUrl + data.picture} alt="" />
						</div>
						<div className="infos-container">
							<time>{(new Date(data.createdAt)).toLocaleString()}</time>
							<p className="description">
								{data.description}
							</p>
						</div>
					</div>
                )
            }
        </section>;
    }

    componentDidMount() {
        this.props.dispatch( fetchPost( this.props.match.params.id ) )
    }
}

function mapStateToProps( state ) {
    return {
        post: state.post
    };
}

export default connect( mapStateToProps )( PostDetail );