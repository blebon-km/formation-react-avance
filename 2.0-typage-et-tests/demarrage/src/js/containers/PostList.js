import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import config from 'config';
import { fetchPosts } from '../actions';

class PostList extends React.Component {
    constructor(...args) {
        super(...args);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    render() {
        return (
            <section className={`post-list ${this.props.posts.isLoading ? 'is-loading' : ''}`}>
                <div className="search-form">
                    <form className="form-inline" onSubmit={this.handleSubmit}>
                        <input className="form-control mr-sm-2" type="text" placeholder="hashtag" />
                        <button className="btn btn-secondary" type="submit">Rechercher</button>
                    </form>
                </div>
                <ul>
                    {this.props.posts.data.map( post => (
                        <li key={post.id}>
                            <Link to={`/posts/${post.id}`}>
                                <article style={{ backgroundImage: `url(${config.picturesUrl + post.picture})`}}>
                                </article>
                            </Link>
                        </li>
                    ) )}
                </ul>
            </section>
        );
    }

    componentDidMount() {
        this.props.dispatch( fetchPosts() )
    }

    handleSubmit( event ) {
        event.preventDefault();
        this.props.dispatch( fetchPosts( event.currentTarget[0].value ) );
    }
}

function mapStateToProps( state ) {
    return {
        posts: state.posts
    };
}

export default connect( mapStateToProps )( PostList );