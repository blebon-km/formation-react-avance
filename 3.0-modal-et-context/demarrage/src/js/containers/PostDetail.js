// @flow

import React from 'react';
import { connect } from 'react-redux';
import config from 'config';
import { fetchPost } from '../actions';
import type { State as PostState } from '../reducer/post';

type Props = {
    post: PostState,
    dispatch: Function,
    match: {
        params: {
            id: string
        }
    }
}

export class PostDetail extends React.Component<Props> {
    render() {
        const { data, isLoading } = this.props.post;
        return <section className={`post-detail ${isLoading ? "is-loading" : ""}`}>
            {
                data && (
                    <div>
                        <div className="picture-container">
                            <img src={config.picturesUrl + data.picture} alt="" />
                        </div>
                        <div className="infos-container">
                            <time>{(new Date(data.createdAt)).toLocaleString()}</time>
                            <p className="description">
                                {data.description}
                            </p>
                        </div>
                    </div>
                )
            }
        </section>;
    }

    componentDidMount() {
        this.props.dispatch( fetchPost( parseInt( this.props.match.params.id ) ) )
    }
}

function mapStateToProps( state ) {
    return {
        post: state.post
    };
}

export default connect( mapStateToProps )( PostDetail );