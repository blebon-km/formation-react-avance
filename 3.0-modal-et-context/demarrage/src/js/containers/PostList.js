//@flow

import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import config from 'config';
import { fetchPosts } from '../actions';
import type { State as PostsState } from '../reducer/posts';

type Props = {
    posts: PostsState,
    dispatch: Function
}

export class PostList extends React.Component<Props> {
    searchInput: ?HTMLInputElement
    handleSubmit: Function

    constructor(...args: any[]) {
        super(...args);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    render() {
        return (
            <section className={`post-list ${this.props.posts.isLoading ? 'is-loading' : ''}`}>
                <div className="search-form">
                    <form className="form-inline" onSubmit={this.handleSubmit}>
                        <input className="form-control mr-sm-2" name="search" type="text" placeholder="hashtag" ref={ elem => this.searchInput = elem} />
                        <button className="btn btn-secondary" type="submit">Rechercher</button>
                    </form>
                </div>
                <ul>
                    {this.props.posts.data.map( post => (
                        <li key={post.id}>
                            <Link to={`/posts/${post.id}`}>
                                <article style={{ backgroundImage: `url(${config.picturesUrl + post.picture})`}}>
                                </article>
                            </Link>
                        </li>
                    ) )}
                </ul>
            </section>
        );
    }

    componentDidMount() {
        this.props.dispatch( fetchPosts() )
    }

    handleSubmit( event: Event ) {
        event.preventDefault();
        this.props.dispatch(fetchPosts( this.searchInput && this.searchInput.value ) );
    }
}

function mapStateToProps( state ) {
    return {
        posts: state.posts
    };
}

export default connect( mapStateToProps )( PostList );