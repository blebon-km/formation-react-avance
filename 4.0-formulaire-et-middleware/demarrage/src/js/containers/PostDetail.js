// @flow

import React from 'react';
import { connect } from 'react-redux';
import { push } from 'connected-react-router';
import { fetchPost } from '../actions';
import type { State as PostState } from '../reducer/post';
import Modal from '../components/Modal';
import ConfigContext from "../contexts/ConfigContext";

type Props = {
    post: PostState,
    dispatch: Function,
    match: {
        params: {
            id: string
        }
    }
}

export class PostDetail extends React.Component<Props> {
    static contextType = ConfigContext

    render() {
        const { data, isLoading } = this.props.post;
        return (
            <Modal onClick={(event) => { this.props.dispatch( push( '/' )); console.log( event.target, event.currentTarget ); }}>
                <section className={`post-detail ${isLoading ? "is-loading" : ""}`}>
                    {
                        data && (
                            <div>
                                <div className="picture-container">
                                    <img src={this.context.picturesUrl + data.picture} alt="" />
                                </div>
                                <div className="infos-container">
                                    <time>{(new Date(data.createdAt)).toLocaleString()}</time>
                                    <p className="description">
                                        {data.description}
                                    </p>
                                </div>
                            </div>
                        )
                    }
                </section>          
            </Modal>
        )
        
    }

    fetchPost() {
        this.props.dispatch( fetchPost( parseInt( this.props.match.params.id ) ) )
    }

    componentDidUpdate( prevProps: Props ) {
        if ( prevProps.match.params.id !== this.props.match.params.id )
        {
            this.fetchPost();
        }
    }

    componentDidMount() {
        this.fetchPost();
    }
}

function mapStateToProps( state ) {
    return {
        post: state.post
    };
}

export default connect( mapStateToProps )( PostDetail );