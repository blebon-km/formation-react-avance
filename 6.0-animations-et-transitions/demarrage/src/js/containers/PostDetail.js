// @flow

import React from 'react';
import { connect } from 'react-redux';
import { push } from 'connected-react-router';
import { reduxForm, Field } from 'redux-form';
import {Map,List} from 'immutable';
import { fetchPost, addComment } from '../actions';
import type { State as PostState } from '../reducer/post';
import Modal from '../components/Modal';
import ConfigContext from "../contexts/ConfigContext";

type Props = {
    post: PostState,
    dispatch: Function,
    match: {
        params: {
            id: string
        }
    },
    handleSubmit: Function
}

export class PostDetail extends React.Component<Props> {
    static contextType = ConfigContext

    render() {
		const data:?Map<string,any> = this.props.post.get('data'),
			filter:?string = data ? data.get('filter') : null,
			picture:?string = data ? data.get('picture') : null,
			createdAt:?string = data ? data.get('createdAt') : null,
			description:?string = data ? data.get('description') : null,
			comments:?List<any> = data ? data.get('comments') : null,
			isLoading:?boolean = this.props.post.get('isLoading');
		return (
            <Modal onClick={() => { this.props.dispatch( push( '/' )) }}>
                <section className={`post-detail ${isLoading ? "is-loading" : ""}`}>
                    {
                        data && (
                            <div>
                                <div className="picture-container">
                                    <img className={ filter && `filter-${filter}` } src={this.context.picturesUrl + picture} alt="" />
                                </div>
                                <div className="infos-container">
                                    <time>{createdAt && new Date(createdAt).toLocaleString()}</time>
                                    <p className="description">
                                        {description}
                                    </p>
                                    <form onSubmit={this.props.handleSubmit}>
                                        <div className="form-group">
                                            <Field
                                                component="input"
                                                name="nickname"
                                                type="text"
                                                className="form-control form-control-sm"
                                                placeholder="Votre pseudo"
                                            />
                                        </div>
                                        <div className="form-group">
                                            <Field
                                                component="textarea"
                                                name="content"
                                                className="form-control form-control-sm"
                                                placeholder="Votre commentaire..."
                                            />
                                        </div>
                                        <div className="form-group">
                                            <button type="submit" className="btn btn-primary btn-sm">Ajouter</button>
                                        </div>
                                    </form>
                                    <ul className="comment-list">
                                        {comments && comments.map( comment => (
                                            <li key={comment.id}>
                                                <p>
                                                    <strong>{comment.get('nickname')}</strong>{' '}
                                                    {comment.get('content')}
                                                </p>
                                            </li>
                                        ) )}
                                    </ul>
								</div>
                            </div>
						)
                    }
                </section>
            </Modal>
        )

    }

    fetchPost() {
        this.props.dispatch( fetchPost( parseInt( this.props.match.params.id ) ) )
    }

    componentDidUpdate( prevProps: Props ) {
        if ( prevProps.match.params.id !== this.props.match.params.id )
        {
            this.fetchPost();
        }
    }

    componentDidMount() {
        this.fetchPost();
    }
}

function mapStateToProps( state ) {
    return {
        post: state.post
    };
}

PostDetail = connect( mapStateToProps )( PostDetail );
PostDetail = reduxForm({
    form: 'comment',
    onSubmit: (values, dispatch, props) => {
        const id: number = parseInt( props.match.params.id );
        return dispatch( addComment( id, values.content, values.nickname ) )
            .then( () => dispatch( fetchPost( id ) ) );
    },
})( PostDetail );

export default PostDetail;
