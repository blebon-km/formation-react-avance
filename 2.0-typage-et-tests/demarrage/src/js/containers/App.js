import React from 'react';
import { Switch, Route, withRouter } from 'react-router';
import Menu from '../components/Menu';
import PostList from './PostList';
import PostDetail from './PostDetail';

class App extends React.Component {
    render() {
        return (
            <div>
                <Menu />
                <div className="container">
                    <Switch>
                        <Route exact path="/" component={PostList} />
                        <Route exact path="/posts/:id" component={PostDetail} />
                    </Switch>
                </div>
            </div>
        )
    }
}

export default withRouter( App );