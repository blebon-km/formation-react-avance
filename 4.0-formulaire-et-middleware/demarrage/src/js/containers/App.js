import React from 'react';
import { Route, withRouter } from 'react-router';
import Menu from '../components/Menu';
import PostList from './PostList';
import PostDetail from './PostDetail';

class App extends React.Component {
    render() {
        return (
            <div>
                <Menu />
                <div className="container">
                    <Route path="/" component={PostList} />
                    <Route exact path="/posts/:id" component={PostDetail} />
                </div>
            </div>
        )
    }
}

export default withRouter( App );