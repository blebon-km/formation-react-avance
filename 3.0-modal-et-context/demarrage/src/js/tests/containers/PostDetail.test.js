jest.mock('config', () => {
    return {
        basePath: '/',
        baseUrl: 'http://localhost/',
        apiUrl: 'http://localhost/api/',
        picturesUrl: 'http://localhost/uploads/pictures/',
    };
}, { virtual: true });

import React from 'react';
import renderer from "react-test-renderer";
import { PostDetail } from '../../containers/PostDetail';

describe( 'PostDetail', function() {
    it( 'should render correctly', function() {
        const tree1 = renderer
            .create(<PostDetail dispatch={() => {}} match={{ params: { id: 42 } }} post={{ data: null, isLoading: true, error: null }} />)
            .toJSON();
        expect(tree1).toMatchSnapshot();

        const tree2 = renderer
            .create(<PostDetail dispatch={() => {}} match={{ params: { id: 42 } }} post={{ data: null, isLoading: false, error: { message: 'An error occured' } }} />)
            .toJSON();
        expect(tree2).toMatchSnapshot();  

        const tree3 = renderer
            .create(<PostDetail dispatch={() => {}} match={{ params: { id: 42 } }} post={{ data: {
                id: 42,
                picture: 'picture.jpg',
                description: 'My picture #cool',
                createdAt: '2018-11-28T16:21:01+00:00',
                comments: [ { id: 1, content: 'Great post !', nickname: 'John', createdAt: '2018-11-28T16:21:01+00:00'} ]
            }, isLoading: false, error: null }} />)
            .toJSON();
        expect(tree3).toMatchSnapshot();  
    } )
} )