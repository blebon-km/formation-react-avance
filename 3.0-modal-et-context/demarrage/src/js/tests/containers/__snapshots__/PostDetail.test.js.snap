// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetail should render correctly 1`] = `
<section
  className="post-detail is-loading"
/>
`;

exports[`PostDetail should render correctly 2`] = `
<section
  className="post-detail "
/>
`;

exports[`PostDetail should render correctly 3`] = `
<section
  className="post-detail "
>
  <div>
    <div
      className="picture-container"
    >
      <img
        alt=""
        src="http://localhost/uploads/pictures/picture.jpg"
      />
    </div>
    <div
      className="infos-container"
    >
      <time>
        2018-11-28 17:21:01
      </time>
      <p
        className="description"
      >
        My picture #cool
      </p>
    </div>
  </div>
</section>
`;
